OmiseSDK/AtomeFormViewController.swift

```diff
@@ -0,0 +1,241 @@
+        applySecondaryColor()
+        navigationItem.backBarButtonItem = UIBarButtonItem(title: "", style: .plain, target: nil, action: nil)
+        
+        formFields.forEach {
+            $0.inputAccessoryView = formFieldsAccessoryView
+        }
+        
+        formFields.forEach {
+            $0.adjustsFontForContentSizeCategory = true
+        }
+        formLabels.forEach {
+            $0.adjustsFontForContentSizeCategory = true
+        }
+        submitButton.titleLabel?.adjustsFontForContentSizeCategory = true
+        
+        if  #available(iOS 11, *) {
+            // We'll leave the adjusting scroll view insets job for iOS 11 and later to the layoutMargins + safeAreaInsets here
+        } else {
+            automaticallyAdjustsScrollViewInsets = true
+        }
+
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(keyboardWillChangeFrame(_:)),
+            name: NotificationKeyboardWillChangeFrameNotification,
+            object: nil
+        )
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(keyboardWillHide(_:)),
+            name: NotificationKeyboardWillHideFrameNotification,
+            object: nil
+        )
+
+        phoneNumberTextField.validator = try? NSRegularExpression(pattern: "\\d{10,11}\\s?", options: [])
+    }
+    
+    override func viewWillLayoutSubviews() {
+        super.viewWillLayoutSubviews()
+        
+        if #available(iOS 11, *) {
+            // There's a bug in iOS 10 and earlier which the text field's intrinsicContentSize is returned the value
+            // that doesn't take the result of textRect(forBounds:) method into an account for the initial value
+            // So we need to invalidate the intrinsic content size here to ask those text fields to calculate their
+            // intrinsic content size again
+        } else {
+            formFields.forEach {
+                $0.invalidateIntrinsicContentSize()
+            }
+        }
+    }
+   
+    @IBAction private func submitForm(_ sender: AnyObject) {
+        guard let name = nameTextField.text?.trimmingCharacters(in: CharacterSet.whitespaces) else {
+            return
+        }
+        
+        guard let email = emailTextField.text?.trimmingCharacters(in: CharacterSet.whitespaces) else {
+            return
+        }
+        
+        guard let phone = phoneNumberTextField.text?.trimmingCharacters(in: CharacterSet.whitespaces) else {
+            return
+        }
+        
+        guard let street = shippingStreetTextField.text?.trimmingCharacters(in: CharacterSet.whitespaces) else {
+            return
+        }
+        
+        guard let city = shippingCityTextField.text?.trimmingCharacters(in: CharacterSet.whitespaces) else {
+            return
+        }
+        
+        guard let country = shippingCountryCodeTextField.text?.trimmingCharacters(in: CharacterSet.whitespaces) else {
```

Comment:
I'm not sure if we have validation for this mandatory country code field, I remember that our Source API will throw an error if it’s not the 2-length characters 😂

Reply:
Nice question ka, I will add this one too! 🙏